src/config/backends/extlinux-file.ts

```diff
@@ -0,0 +1,166 @@
+ * that we would want to be able to parse (get the value) and generate (create a value).
+ *
+ */
+export abstract class ConfigurableDirective {
+	// Parses the values for this directive
+	public abstract parse(directives: Directive): ConfigOptions;
+	// Return the value to be set for this directive using the provided ConfigOptions
+	public abstract generate(opts: ConfigOptions, existingValue?: string): string;
+}
+
+/**
+ * AppendDirective
+ *
+ * Add one or more options to the kernel command line.
+ *
+ */
+export class AppendDirective extends ConfigurableDirective {
+	private supportedConfigValues: string[];
+
+	public constructor(supportedConfigValues: string[]) {
+		super();
+		this.supportedConfigValues = supportedConfigValues;
+	}
+
+	/**
+	 * Parses a APPEND directive string into a ConfigOptions
+	 *
+	 * Example:
+	 * 	parse({ APPEND: "ro rootwait isolcpus=0,4" })
+	 * 	-> { 'ro': '', 'rootwait': '', 'isolcpus': '0,4' }
+	 *
+	 */
+	public parse(directives: Directive): ConfigOptions {
+		// Check that there is an APPEND directive to parse
+		if (directives.APPEND == null) {
+			throw new ExtLinuxParseError(
+				'Could not find APPEND directive in default extlinux.conf boot entry',
+			);
+		}
+		// Parse all the key and values into ConfigOptions
+		return directives.APPEND.split(' ').reduce(
+			(configOpts: ConfigOptions, appendValue: string) => {
+				// Break this append config into key and value
+				const [KEY, VALUE = '', ...more] = appendValue.split('=');
+				if (!KEY) {
+					return configOpts; // No value to set so return
+				} else if (more.length > 0) {
+					// APPEND value is not formatted correctly
+					// Example: isolcpus=3=2 (more then 1 value being set)
+					throw new AppendDirectiveError(
+						`Unable to parse invalid value: ${appendValue}`,
+					);
+				}
+				// Return key value pair with existing configs
+				return { [KEY]: VALUE, ...configOpts };
+			},
+			{},
+		);
+	}
+
+	/**
+	 * Generates a string value for APPEND directive given a ConfigOptions
+	 *
+	 * Keys in existingValue that are also in the provided ConfigOptions
+	 *  will be replaced with those from opts.
+	 *
+	 * Example:
+	 * 	generate({ isolcpus: '0,4' })
+	 * 	-> 'isolcpus=0,4'
+	 *
+	 */
+	public generate(opts: ConfigOptions, existingValue: string = ''): string {
+		// Parse current append line and remove whitelisted values
+		// We remove whitelisted values to avoid duplicates
+		const appendLine = existingValue.split(' ').filter((entry) => {
+			const lhs = entry.split('=');
+			return !this.supportedConfigValues.includes(lhs[0]);
+		});
+		// Add new configurations values to the provided append line
+		return appendLine
+			.concat(
+				_.map(opts, (value, key) => {
+					if (key.includes('=') || value.includes('=')) {
+						throw new AppendDirectiveError(
+							`One of the values being set contains an invalid character: [ value: ${value}, key: ${key} ]`,
+						);
+					} else if (!value) {
```

Comment:
What are you looking for with `!value`? If it can be null then the `value.includes('=')` would need to be `value?.includes('=')`

Reply:
good catch. I had thought includes would return false since that character is not included in null/undefined but actually includes is only for string prototype so it errors...